File: MEL/Aufgabe1Csharp/Logik/Town.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Aufgabe1Csharp.Logik
{
    class Town
    {
        public readonly int x;
        public readonly int y;
        public String name;

        public Town(int x = 0, int y = 0, string name = "no Name")
        {
            this.x = x;
            this.y = y;
            this.name = name;
        }

        public double distance(ref Town t)
        {
            int deltax = Math.Abs(x - t.x);
            int deltay = Math.Abs(y - t.y);
            return Math.Sqrt(deltax * deltax + deltay * deltay);
        }
    }
}

File: MEL/MEL/Logik/Town.vb
Imports System.Math

Public Class Town

#Region "------------------------------------ fields ------------------------------------"
    Private xCoord As Integer
    Private yCoord As Integer
#End Region

#Region "------------------------------------ properties ------------------------------------"
    Public ReadOnly Property x As Integer
        Get
            Return xCoord
        End Get
    End Property

    Public ReadOnly Property y As Integer
        Get
            Return yCoord
        End Get
    End Property

    Public Property name As String
#End Region
#Region "------------------------------------ initialisation ------------------------------------"
    Sub New(Optional x As Integer = 0, Optional y As Integer = 0, Optional name As String = "noname")
        xCoord = x
        yCoord = y
        Me.name = name
    End Sub

#End Region

    Public Function distance(ByRef s As Town) As Double
        Dim deltaX = Abs(xCoord - s.xCoord)
        Dim deltaY = Abs(yCoord - s.yCoord)
        Return Sqrt(deltaX * deltaX + deltaY * deltaY)
    End Function
End Class
